### PR TITLE
fix(core): make sure that an iframe sanitizer always uses the right value and returns the result

### PR DESCRIPTION
**Pull request by AndrewKushnir**
This commit updates the iframe sanitizer function (the `ɵɵvalidateIframeAttribute` one) to resolve a couple issues related to the fact that the value was not returned from the sanitizer as well as in some cases a wrong tNode was used as a source of info.

## PR Type
What kind of change does this PR introduce?

- [x] Bugfix

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: packages/core/src/sanitization/iframe_attrs_validation.ts [hunk 1]
```diff
@@ -35,15 +35,25 @@ export const IFRAME_SECURITY_SENSITIVE_ATTRS = new Set(
...
   const element = getNativeByTNode(tNode, lView) as RElement | RComment;
 
   if (tNode.type === TNodeType.Element && tNode.value.toLowerCase() === 'iframe' &&
       // Note: check for all false'y values including an empty string as a value,
       // since this is a default value for an `<iframe>`'s `src` attribute.
       ((element as HTMLIFrameElement).src || (element as HTMLIFrameElement).srcdoc)) {
+    // Unset `src` and/or `srcdoc` if we come across a situation when
+    // a security-sensitive attribute is applied later.
+    const iframe = element as HTMLIFrameElement;
+    if (iframe.src) {
```
Comment (pkozlowski-opensource) [inline packages/core/src/sanitization/iframe_attrs_validation.ts:48]: I don't think we need to have this check - at this point we've detected a problematic pattern, right? So no need to optimize further processing

File: packages/core/test/acceptance/security_spec.ts [hunk 2]
```diff
@@ -512,7 +526,76 @@ describe('iframe processing', () => {
...
+           }
+
+           expectIframeToBeCreated(IframeComp, {
+             src: TEST_IFRAME_URL,
+             referrerPolicy: 'no-referrer',
+           });
+         });
+
+      ['src', 'srcdoc'].forEach((srcAttr: string) => {
+        it('should work when a security-sensitive attribute is set before `src` ' +
```
Comment (alxhub) [inline packages/core/test/acceptance/security_spec.ts:550]:
```suggestion
        it(`should work when a security-sensitive attribute is set before '${srcAttr}' ` +
```

**Review by alxhub — APPROVED** [review]
Reviewed-for: fw-security

**Comment by AndrewKushnir** [comment]
Note to self: we'd need to incorporate this fix from Bjarki as well to make the code compatible with Trusted Typed:
```
diff --git a/packages/core/src/sanitization/iframe_attrs_validation.ts b/packages/core/src/sanitization/iframe_attrs_validation.ts
--- a/packages/core/src/sanitization/iframe_attrs_validation.ts
+++ b/packages/core/src/sanitization/iframe_attrs_validation.ts
@@ -13,6 +13,7 @@ import {RComment, RElement} from '../ren
 import {LView} from '../render3/interfaces/view';
 import {getCurrentTNode, getLView, getSelectedTNode} from '../render3/state';
 import {getNativeByTNode} from '../render3/util/view_utils';
+import {trustedHTMLFromString} from '../util/security/trusted_types';
 
 /*
  * The set of security-sensitive attributes of an `<iframe>` that *must* be
@@ -49,7 +50,7 @@ export function ɵɵvalidateIframeAttribute(attrValue: any, tagName: string, attrName: string) {
       iframe.src = '';
     }
     if (iframe.srcdoc) {
-      iframe.srcdoc = '';
+      iframe.srcdoc = trustedHTMLFromString('') as unknown as string;
     }
     throw unsafeIframeAttributeError(lView, attrName);
   }
```

**Pull request closed without merging**